src/mmresource.ts

```diff
@@ -0,0 +1,71 @@
+import { KubeConfig, CustomObjectsApi } from '@kubernetes/client-node';
+
+export interface AllocationResponse {
+    kind: string;
+    apiVersion: string;
+    metadata: {
+        name: string;
+        namespace: string;
+        creationTimestamp: string;
+    }
+    spec: Record<string, unknown>;
+    status: {
+        state: string;
+        gameServerName: string;
+        ports: [
+            {
+                name: string;
+                port: number;
+            },
+        ]
+        address: string;
+        nodeName: string;
+    }
+}
+
+export interface ServerRecord {
+    ip: string;
+    port: number;
+    serverName: string;
+}
+
+export class MatchMakingServerAllocator {
+    api: CustomObjectsApi;
+
+    fleet: string;
+
+    namespace: string;
+
+    constructor(fleet: string, namespace: string) {
+        const kc: KubeConfig = new KubeConfig();
+        kc.loadFromDefault();
+        this.api = kc.makeApiClient(CustomObjectsApi);
+
+        this.fleet = fleet;
+        this.namespace = namespace;
+    }
+
+    async allocateServer() : Promise<ServerRecord|null> {
+        const res = await this.api.createNamespacedCustomObject('allocation.agones.dev', 'v1', this.namespace, 'gameserverallocations', {
+            kind: 'GameServerAllocation',
+            spec: {
+                required: {
+                    matchLabels: {
+                        'agones.dev/fleet': this.fleet,
+                    },
+                },
+            },
+        });
+
+        const alloc : AllocationResponse = <AllocationResponse> res.body;
```

Comment:
Can `res` be a failure? What happens here if `this.api.createNamespacedCustomObject` cannot contact the kubernetes API, does matchmaking just crash?

Reply:
If it can't contact the cluster, yeah it crashes. I wasn't actually sure if there would be a way to recover from this, since the matchmaking service can't actually serve any matches at that point. I'd be open to suggestions.

Reply:
Resolved in slack, crash considered acceptable in this case